### ISSUE TITLE
Not an issue - question regarding code

### ISSUE THREAD

**Issue by RichardMisencik** [fpatseas/in-stock-notifier#1]
Not an issue, hope it's okay to ask here, i'm beginner in python but can someone please explain what is the purpose of this code from core.py starting at line 40? i understand the first if statement but why the second one does the same thing (send alert)?:

```
if len(found) > 0 and bool(product["notifyiffound"]) == True:
    send_email = True
elif len(found) == 0 and bool(product["notifyiffound"]) == False:
    send_email = True
```

**Comment by fpatseas** [comment]
Hello @RichardMisencik
You are right. This statement could be refactored into a single if statement. Please do submit a pull request, and I will be more than happy to merge your change.

**Comment by RichardMisencik** [comment]
Hello @fpatseas
Thanks for reply, if u dont mind i have a follow up question, im still not sure if i understand the elif statement correctly, it being there is a mistake or was it a design decision that has meaning that i'm missing? because it is checking if **notify if found** is enabled for current product even if **stock keyword** was not found and then sends email.
Sorry to bother and bad explanation.

**Comment by fpatseas** [comment]
Hello @RichardMisencik
You seem a little confused, so let me explain how this scenario works. The goal is to get notified when a product is available again. In some cases, it is easier to categorize a product as available without having an object (e.g. a button) on the page than if the object does exist. For example, it is easier to look for a non-existent sold-out button or an unavailable label than for a purchase now or an add to cart button. I've implemented it this way to make the script more flexible. The if statement you are referring to evaluates exactly these cases. Do I need to notify if I found the item or not? In other words, and about your previous question, a simple OR operator would simplify this statement, as this would eliminate the need for an else statement. I hope that makes sense to you.
